src/current/v23.1/sso-sql.md

```diff
@@ -1,158 +1,257 @@
-{{site.data.alerts.end}}
+    By default, your cluster's configuration will contain the {{ site.data.products.db }}'s own public key, allowing {{ site.data.products.db }} to serve as an IdP. When modifying this cluster setting, you must include the {{ site.data.products.db }} public key in the key set. Failing to do so will lock out console SSO users and prevent maintenance access by the {{ site.data.products.db }} managed service, leading to unintended consequences.
+
+    {{site.data.alerts.callout_success}}
+    {{ site.data.products.db }}'s IdP configuration can be viewed publicly at: [`https://cockroachlabs.cloud/.well-known/openid-configuration`](https://cockroachlabs.cloud/.well-known/openid-configuration).
+    The link specified for `jwks_uri` provides the IdP's public signing key. For {{ site.data.products.db }} this is `https://cockroachlabs.cloud/oauth2/keys`
+    {{site.data.alerts.end}}
+
+    For example:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    SET CLUSTER SETTING server.jwt_authentication.jwks = '{
+      "keys": [
+        {
+          "alg": "RS256",
+          "kty": "RSA",
+          "kid": "85ba9313fd7a7d4afa84884abcc8403004363180",
+          "e": "AQAB",
+          "use": "sig",
+          "n": "pP-rCe4jkKX6mq8yP1GcBZcxJzmxKWicHHor1S3Q49u6Oe-bQsk5NsK5mdR7Y7liGV9n0ikXSM42dYKQdxbhKA-7--fFon5isJoHr4fIwL2CCwVm5QWlK37q6PiH2_F1M0hRorHfkCb4nI56ZvfygvuOH4LIS82OzIgmsYbeEfwDRpeMSxWKwlpa3pX3GZ6jG7FgzJGBvmBkagpgsa2JZdyU4gEGMOkHdSzi5Ii-6RGfFLhhI1OMxC9P2JaU5yjMN2pikfFIq_dbpm75yNUGpWJNVywtrlNvvJfA74UMN_lVCAaSR0A03BUMg6ljB65gFllpKF224uWBA8tpjngwKQ"
+        },
+        {
+          "alg": "RS256",
+          "n": "8ASqcDDTYmFx53uAkWlmqx8qIx0WQacaswAAb7xGx9XL2T71VPtKKgPqoimXzj4fXT2F3opIgOQgGKxkxw2QrAAoejdIud5URrZLRponmbMQO3erG5sd8PC29rjekX8_hvX_AMT3zU8JLKVLFR0xJmFdmIomnarpEvWKmG0SNvbuVveprsve0n1W35uodcRe417vlTNnH9j8fesQekvFf8tIWbHouXDg3B1km4gqZBbQ_MWvGriGBY5sfr7A2d0iNe89Aje7pz1RFLUFOu-u_NyD6RwL6qo4_yetAYIzm02a2KAAq03YPs2LHMVQmkh1LtyeuA6bvf9146cAFY4BCQ",
+          "use": "sig",
+          "e": "AQAB",
+          "kty": "RSA",
+          "kid": "05150a1320b9395b05716877376928509bab44ac"
+        }
+      ]
+    }';
+    ~~~
+
+1. Configure token generation.
 
 {% include_cached copy-clipboard.html %}
 ~~~shell
-cockroach sql --url "postgresql://{SQL_USERNAME}:{JWT_TOKEN}@{CLUSTER_HOST}:26257?options=--crdb:jwt_auth_enabled=true" --certs-dir={CLUSTER_CERT_DIR}
+SET CLUSTER SETTING server.oidc_authentication.generate_cluster_sso_token.enabled = true;
+SET CLUSTER SETTING server.oidc_authentication.generate_cluster_sso_token.use_token = id_token;
 ~~~
 
-~~~txt
-Welcome to the cockroach SQL interface...
-~~~
+## How CockroachDB determines the SQL username from a JWT
+
+1. `server.oidc_authentication.generate_cluster_sso_token.use_token` determines which token to look at (`id_token` or `access_token`).
```

Comment:
I think you are confusing two different things in this section. Cluster SSO just uses the claim and the identity_map. This cluster setting just tells the UI which token to send to you which you assert has all the right claims for cluster SSO.

Reply:
I'm seeing "The `use_token` field determines which part of the received OIDC credentials the cluster uses to [determine the user’s identity](#how-cockroachdb-determines-the-sql-username-from-a-jwt)." now which doesn't seem quite right. It's which token gets displayed to a user.

Reply:
Hmm, I definitely need more help with "How CockroachDB determines the SQL username from a JWT", is this incorrect?
1. `server.oidc_authentication.generate_cluster_sso_token.use_token` determines which token to look at (`id_token` or `access_token`).
1. `server.jwt_authentication.claim` determines which claim in that token, defaulting to picking the token’s subject, but also potentially using email.
1. `server.identity_map.configuration` maps that claim (along with the token’s issuer) to a SQL username.

Please clarify, thanks

Reply:
@kpatron-cockroachlabs just wanted to follow up here. I think if the current copy is good I'm ready to merge, but it seems like I might have some misinformation here?

Reply:
Use_token is only relevant to the `generate_cluster_sso_token` flow and so while it's kinda true that it flows through the claim and configuration, that is only relevant for the thing that is generated to the user, NOT anything that directly logs in.